Decide to read from stdin using a parameter rather than magic

loren138 opened the pull request:
## Proposed changes

This changes webdriverio to read from stdin if `--stdin` or `-` is provided as an option as a fix to #2624. (Note: This is backwards breaking of #2617 but restores functionality closer to #2477.  The conversation in #2624 seemed to decide a new option was the best way to implement this.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works (see comments below)
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

I could not find any tests which exercise `cli.js` so I didn't add any tests, but let me know if there is a spot to add them.

The single dash `-` is read in as a file (`argv._`) rather than an option so I had to alter the reading of the config to ensure it didn't try to read in dash as the config file.

### Reviewers: @christian-bromann

loren138 commented:
My build seems to have failed due to a sauce-connect failure.  Is there something I should change to fix that?

loren138 commented:
I signed the agreement.  It shows in the build status, but the comment didn't get updated for some reason.

BorisOsipov commented:
@loren138 check your `git config` do you perform commits with email your used to sign CLA?

loren138 commented:
@BorisOsipov I was committing with the wrong email originally.  I've fixed it now.  I went ahead and did a rebase to fix the author.  `licence/cla` has a green check in the status checks saying I signed it.  Is there a chance it just isn't going to update the comment?  (Maybe because I fixed authors with a rebase?)

BorisOsipov commented:
@loren138 
>licence/cla has a green check in the status

it is enough. don't pay attention to `jsf-clabot ` comments

jana9281 commented:
Confirmed that fix is working on my test framework - running tests using Webstorm AND GitBash

sgreener commented:
This solves my issue to do with being able to connect to the debugger (in 4.12.0)
